conda_env/cli/main_create.py

```diff
@@ -51,22 +53,36 @@ def configure_parser(sub_parsers):
         '-q', '--quiet',
         default=False,
     )
+    p.add_argument(
+        'handle',
+        help='environment definition\'s handle',
+        default=''
+    )
     common.add_parser_json(p)
     p.set_defaults(func=execute)
 
 
 def execute(args, parser):
-    try:
-        env = from_file(args.file)
-    except exceptions.EnvironmentFileNotFound as e:
-        msg = 'Unable to locate environment file: %s\n\n' % e.filename
-        msg += "\n".join(textwrap.wrap(textwrap.dedent("""
-            Please verify that the above file is present and that you have
-            permission read the file's contents.  Note, you can specify the
-            file to use by explictly adding --file=/path/to/file when calling
-            conda env create.""").lstrip()))
-
-        common.error_and_exit(msg, json=args.json)
+
+    if args.handle:
+        loader = get_loader(args.handle)
+        if loader is not None:
+            env = from_yaml(loader.get(), args.name)
+        else:
+            raise exceptions.LoaderNotFound(args.handle)
+
+    else:
+        try:
+            env = from_file(args.file)
+        except exceptions.EnvironmentFileNotFound as e:
+            msg = 'Unable to locate environment file: %s\n\n' % e.filename
+            msg += "\n".join(textwrap.wrap(textwrap.dedent("""
+                Please verify that the above file is present and that you have
+                permission read the file's contents.  Note, you can specify the
+                file to use by explictly adding --file=/path/to/file when calling
+                conda env create.""").lstrip()))
+
+            common.error_and_exit(msg, json=args.json)
```

Comment:
Unrelated, but if you wanted to do some code weeding, this should be an exception as well.